### PR TITLE
Fdl unit tests

### PR DESCRIPTION
**Pull request by eldhosembabu**
FDL Custom domain validation logic change.

- We were doing regex search because of which the regex might sometime matches based on the contents of the url present in the link query parameter. Now this is changed to start to end validation with. "^" and "$".
- We were not honoring the order of link query param in the url before as we were looking the link param to be the first param in the url. Now this got changed to accept link param anywhere in the url.

Issue : https://github.com/firebase/firebase-ios-sdk/issues/6978

### CODE REVIEW

File: FirebaseDynamicLinks/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# v8.10.0
```
Comment (paulb777) [inline FirebaseDynamicLinks/CHANGELOG.md:1]: Next release is 8.15.0. See https://github.com/firebase/firebase-ios-sdk/milestones

**Review by paulb777 — COMMENTED** [review]
Looks like GitHub CI triggering is currently broken.

Have you confirmed the test breaks locally without the fix and passes with it?

File: FirebaseDynamicLinks/CHANGELOG.md (hunk 1 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/CHANGELOG.md]: Done

**Review by paulb777 — APPROVED** [review]

**Comment by eldhosembabu** [comment]
Modified logic to handle @"https://google.com?some=qry&somelink=https%3A%2F%2Fsomedomain" pointed out by 3p dev in the related issue.

**Review by paulb777 — APPROVED** [review]
Much more readable than the regex and great to add the new tests. Thanks!

**Pull request merged**